apollo-federation-types/src/config/version.rs

```diff
@@ -183,3 +185,100 @@ impl FromStr for FederationVersion {
         }
     }
 }
+
+impl<'de> Deserialize<'de> for FederationVersion {
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    where
+        D: Deserializer<'de>,
+    {
+        struct Visitor;
+
+        impl<'de> serde::de::Visitor<'de> for Visitor {
+            type Value = FederationVersion;
+
+            fn expecting(&self, f: &mut fmt::Formatter) -> fmt::Result {
+                f.write_str("literal '1' or '2' (as a string or number), or a fully qualified version prefixed with an '=', like: =2.0.0")
+            }
+
+            fn visit_u64<E>(self, num: u64) -> Result<Self::Value, E>
+            where
+                E: Error,
+            {
+                match num {
+                    0 | 1 => Ok(FederationVersion::LatestFedOne),
+                    2 => Ok(FederationVersion::LatestFedTwo),
+                    _ => Err(Error::custom(format!(
+                        "specified version `{}` is not supported",
+                        num
+                    ))),
```

Comment:
An alternative approach is to convert int to string and use `from_str`, which is more _consistent_, but resource inefficient. Either way is fine with me